dali/kernels/signal/resampling.h

```diff
@@ -116,9 +135,25 @@ struct Resampler {
         if (i1 + in_block_i >= n_in)
           i1 = n_in - 1 - in_block_i;
         float f = 0;
-        float x = i0 - in_pos;
-        for (int i = i0; i <= i1; i++, x++) {
-          assert(in_block_ptr + i >= in && in_block_ptr + i < in + n_in);
+        int i = i0;
+
+#ifdef __SSE2__
+        __m128 f4 = _mm_setzero_ps();
+        __m128 x4 = _mm_setr_ps(i - in_pos, i+1 - in_pos, i+2 - in_pos, i+3 - in_pos);
```

Comment:
Can we assume that we have always at least 4 elements available?

Reply:
If we don't, the vectorized loop won't get executed at all, so we're safe.